### PR TITLE
Place array-valued entity example on the array schema, not items

### PR DESCRIPTION
**Pull request by olivier-thatch**
## Problem

A `grape-entity` exposure declared as an array via `is_array: true` that
also carries an `example:` attaches the array-valued example to the wrong
schema node. The example lands on the inner (scalar) `items` schema, which
is then wrapped as `items` inside the array. The result is a semantically
invalid schema: a scalar `items` whose `example` is the whole array.

## Fix

`apply_exposure_properties` sets the example on the inner schema *before*
`wrap_in_array_if_needed` wraps it as `items`, so the example follows the
inner schema down into `items`. The fix hoists the example back up: when
wrapping an exposure in an array, an **array-valued** example is moved from
the item schema to the array wrapper and cleared from the item schema. A
**scalar** example is left on the item schema, where it remains a valid
example for a single element — so existing behavior for scalar examples is
unchanged.

This is an introspector-layer fix only. It does not touch the exporters.

## Example

```ruby
class AnimalEntity < Grape::Entity
  expose :species, documentation: { type: "string", is_array: true, example: ["cat", "dog"] }
end

class API < Grape::API
  format :json
  desc "list", entity: AnimalEntity
  get("animals") { present({}, with: AnimalEntity) }
end
```

## Schema before / after

OAS 2.0 (`species` property), which renders the array-valued example
end-to-end:

```yaml
# Before — scalar `items` with an array-valued example (invalid)
species:
  type: array
  items:
    type: string
    example:
      - cat
      - dog

# After — example describes the whole array, so it sits on the array node
species:
  type: array
  items:
    type: string
  example:
    - cat
    - dog
```

At the introspector layer (`EntityIntrospector#build_schema`), the same
relocation holds for all three versions: `species.examples == ["cat","dog"]`
and `species.items.examples == nil`. For OAS 3.0/3.1 the array example is
further mangled by a *separate* exporter truncation bug, tracked and fixed
in #104 — that is a different layer and out of scope here.

## Backward compatibility

- Public API surface: unchanged.
- Emitted OAS shape for inputs that don't exercise this fix: unchanged.
  Only entity `is_array:` exposures carrying an **array-valued** example
  change shape — the example moves from `items` up to the array node.
  Scalar examples on `is_array:` exposures are untouched (they stay on
  `items`).
- New runtime/dev dependencies: none.

## Open questions

OAS 3.0/3.1 array-example truncation in the exporter is a separate bug
(#104). This PR is intentionally independent of it and is verified at the
introspector/unit level so it stands alone off `main`.

### CODE REVIEW

File: lib/grape_oas/introspectors/entity_introspector_support/exposure_processor.rb [hunk 1]
```diff
@@ -158,7 +158,19 @@ def wrap_in_array_if_needed(prop_schema, doc)
           is_array = doc[:is_array]
           return prop_schema unless is_array
 
-          ApiModel::Schema.new(type: Constants::SchemaTypes::ARRAY, items: prop_schema)
+          array_schema = ApiModel::Schema.new(type: Constants::SchemaTypes::ARRAY, items: prop_schema)
```
Comment (numbata) [inline lib/grape_oas/introspectors/entity_introspector_support/exposure_processor.rb:161]: Nit: `hoist_array_example` is only called from here and its body is very small. Inlining it makes the array-wrapper behavior easier to read in one place, and reading from `doc[:example]` directly keeps the check and assignment tied to the same source:

```ruby
array_schema = ApiModel::Schema.new(type: Constants::SchemaTypes::ARRAY, items: prop_schema)
if doc[:example].is_a?(Array)
  array_schema.examples = doc[:example]
  prop_schema.examples = nil
end
array_schema
```

The `respond_to?(:examples)` guard can also be dropped because `item_schema` is always an `ApiModel::Schema`, which defines examples.

File: test/grape_oas/introspectors/entity_introspector_array_example_test.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        assert_equal [1, 2], counts.examples
+        assert_nil counts.items.examples
+      end
+
+      # is_array: true with a scalar example — example stays on items (valid per OAS).
+
+      class ScalarExampleOnArrayEntity < Grape::Entity
+        expose :label, documentation: { type: "string", is_array: true, example: "cat" }
+      end
+
```
Comment (numbata) [inline test/grape_oas/introspectors/entity_introspector_array_example_test.rb:38]: There are already `is_array: true` fixtures without `example:` elsewhere, but I don’t see one that asserts the examples behavior directly. Worth adding a small targeted regression test here so the nil-example path is locked down:

```ruby
class NoExampleEntity < Grape::Entity
  expose :tags, documentation: { type: "string", is_array: true }
end

def test_is_array_exposure_without_example_produces_no_examples
  schema = EntityIntrospector.new(NoExampleEntity).build_schema
  tags = schema.properties["tags"]

  assert_equal Constants::SchemaTypes::ARRAY, tags.type
  assert_nil tags.examples
  assert_nil tags.items.examples
end
```

This is less about proving `is_array: true` works generally, and more about ensuring a future guard change does not accidentally put examples on either the array wrapper or its item schema when `example:` is omitted.

**Review by numbata — APPROVED** [review]
The introspector fix looks correct to me and is at the right layer: array-valued examples for `is_array: true` exposures should be attached to the array wrapper, not to the item schema.

One related thing I checked while reviewing: the OAS 3.x exporter currently does `Array(@schema.examples).map { |ex| coerce_example(ex, type) }.first`, so when `@schema.examples` is already a Ruby array like `["cat", "dog"]`, only `"cat"` is rendered. This PR does not introduce that behavior; before this change the example was attached to `items`, and after this change it is attached to the array schema but still hits the existing exporter behavior. Since that is already tracked separately in #104, I’d keep it out of scope for this PR. 👍

File: lib/grape_oas/introspectors/entity_introspector_support/exposure_processor.rb (hunk 1 above)
Comment (numbata) [reply lib/grape_oas/introspectors/entity_introspector_support/exposure_processor.rb:161]: Implemented in 73e26e9: the logic is now inlined and reads directly from `doc[:example]`.

File: test/grape_oas/introspectors/entity_introspector_array_example_test.rb (hunk 2 above)
Comment (numbata) [reply test/grape_oas/introspectors/entity_introspector_array_example_test.rb:38]: Implemented in 73e26e9: added the focused no-example regression test.
